### PR TITLE
Fix incorrect Z direction for AABB's position

### LINKED ISSUE #9598: AABB Confusion
**Your Godot version:**
v4.2.2.stable.mono.official

**Issue description:**
Documentation indicates that the position of an AABB is "usually the corner on the bottom-left and back of the bounding box."

Additionally the documentation indicates that "It's recommended setting the width, height, and depth to non-negative values."

If I'm not mistaken "North" is -Z which would mean if you place the AABB at the bottom-left and back (+Z) of an object an AABB with a positive size will extend behind that object.

Is the "bottom-left and back" section incorrect, or is something else going on?

**URL to the documentation page:**
https://docs.godotengine.org/en/stable/classes/class_aabb.html

### PR DESCRIPTION
**Pull request by Mickeon**
Fixes https://github.com/godotengine/godot-docs/issues/9598

Comes from https://github.com/godotengine/godot/pull/87114 (oops)

### CODE REVIEW

File: doc/classes/AABB.xml [hunk 1]
```diff
@@ -345,14 +345,14 @@
 	</methods>
 	<members>
 		<member name="end" type="Vector3" setter="" getter="" default="Vector3(0, 0, 0)">
-			The ending point. This is usually the corner on the top-right and forward of the bounding box, and is equivalent to [code]position + size[/code]. Setting this point affects the [member size].
+			The ending point. This is usually the corner on the top-right and back of the bounding box, and is equivalent to [code]position + size[/code]. Setting this point affects the [member size].
 		</member>
 		<member name="position" type="Vector3" setter="" getter="" default="Vector3(0, 0, 0)">
-			The origin point. This is usually the corner on the bottom-left and back of the bounding box.
+			The origin point. This is usually the corner on the bottom-left and forward of the bounding box.
```
Comment (Flarkk) [inline doc/classes/AABB.xml:351]: Would `front` be clearer here ?

**Review by AThousandShips — APPROVED** [review]
LGTM, as far as I can tell the changes to fix the incorrect directions are correct (with my correction) and I think the word choice is good

**Review by aaronfranke — COMMENTED** [review]
Please be mindful of the precise terminology. As of Godot 4.1 we have separate concepts of forward/back direction and front/rear sides (Vector3 `MODEL_*` constants). We need to pick one of these wordings and use it consistently:

* Forward -Z and back +Z.
* Front +Z and rear -Z.

However, we should consider just mentioning directly that position is in the negative corner, -X -Y -Z, and end is in the positive corner, +X +Y +Z.

File: doc/classes/AABB.xml [hunk 2]
```diff
@@ -345,14 +345,14 @@
...
 		</member>
 		<member name="position" type="Vector3" setter="" getter="" default="Vector3(0, 0, 0)">
-			The origin point. This is usually the corner on the bottom-left and back of the bounding box.
+			The origin point. This is usually the corner on the bottom-left and front of the bounding box.
 		</member>
 		<member name="size" type="Vector3" setter="" getter="" default="Vector3(0, 0, 0)">
 			The bounding box's width, height, and depth starting from [member position]. Setting this value also affects the [member end] point.
-			[b]Note:[/b] It's recommended setting the width, height, and depth to non-negative values. This is because most methods in Godot assume that the [member position] is the bottom-left-back corner, and the [member end] is the top-right-forward corner. To get an equivalent bounding box with non-negative size, use [method abs].
+			[b]Note:[/b] It's recommended setting the width, height, and depth to non-negative values. This is because most methods in Godot assume that the [member position] is the bottom-left-front corner, and the [member end] is the top-right-back corner. To get an equivalent bounding box with non-negative size, use [method abs].
```
Comment (aaronfranke) [inline doc/classes/AABB.xml:355]: As I mentioned in my comment we shouldn't use front/back, we should either use forward/back or front/rear.

**Review by aaronfranke — CHANGES_REQUESTED** [review]

File: doc/classes/AABB.xml [hunk 3]
```diff
@@ -248,7 +248,7 @@
 			<return type="bool" />
 			<param index="0" name="point" type="Vector3" />
 			<description>
-				Returns [code]true[/code] if the bounding box contains the given [param point]. By convention, points exactly on the right, top, and front sides are [b]not[/b] included.
+				Returns [code]true[/code] if the bounding box contains the given [param point]. By convention, points exactly on the right, top, and forward sides are [b]not[/b] included.
```
Comment (aaronfranke) [inline doc/classes/AABB.xml:251]: This is inconsistent with the rest of this PR.

**Review by aaronfranke — APPROVED** [review]

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**